src/binary/decimal.rs

```diff
@@ -177,19 +140,4 @@ mod binary_decimal_tests {
         assert_eq!(written, expected);
         Ok(())
     }
-
-    #[test]
-    fn oversized_decimal() {
-        let mut buf = vec![];
-        // Even though our output stream is a growable Vec, the encoder uses a
-        // fixed-size, stack-allocated buffer to encode the body of the decimal.
-        // If it's asked to encode a decimal that won't fit (>32 bytes), it should
-        // return an error.
-        let precise_pi = Decimal::new(
-            BigUint::from_str("31415926535897932384626433832795028841971693993751058209749445923078164062862089986280348253421170679").unwrap(),
-            -99
-        );
-        let result = buf.encode_decimal_value(&precise_pi);
-        assert!(result.is_err());
-    }
```

Comment:
This sent me down a useful rabbit hole.

Because I had removed the constructor that accepted a `BigUint` coefficient, it became statically impossible to trigger this case. However, because the `Coefficient` type was a `(UInt, Sign)` tuple, it was possible for it to store magnitudes outside the `i128` range but inside the `u128` range.

In my latest commits I've modified `Coefficient` to an `(Int, Sign)` pair where the `Int`'s sign information always agrees with `Sign` _except_ when the coefficient is a negative zero. This is essentially the same logic as before but with a constructor that's now capped at `Int::MAX`. This had the nice side effect of also eliminating the need for constructors that take an explicit `Sign` parameter because that would only ever be necessary to construct a `-0`. There are now `Coefficient::new(some_int_value)` and `Coefficient::negative_zero()` constructors.

All of that too say, it's now statically impossible to programmatically construct a `Decimal` that's too big.